Make the "Arm Versatile" board even more versatile, for QEMU, which can
stick weird processors into things that were never rmeant to receive them.

diff -ru linux/arch/arm/mm/Kconfig linux2/arch/arm/mm/Kconfig
--- linux/arch/arm/mm/Kconfig	2009-12-02 21:51:21.000000000 -0600
+++ linux2/arch/arm/mm/Kconfig	2009-12-08 04:29:50.000000000 -0600
@@ -111,7 +111,7 @@
 
 # ARM920T
 config CPU_ARM920T
-	bool "Support ARM920T processor" if ARCH_INTEGRATOR
+	bool "Support ARM920T processor" if ARCH_INTEGRATOR || ARCH_VERSATILE_PB || ARCH_VERSATILE_AB
 	select CPU_32v4T
 	select CPU_ABRT_EV4T
 	select CPU_PABRT_LEGACY
@@ -129,7 +129,7 @@
 
 # ARM922T
 config CPU_ARM922T
-	bool "Support ARM922T processor" if ARCH_INTEGRATOR
+	bool "Support ARM922T processor" if ARCH_INTEGRATOR || ARCH_VERSATILE_PB || ARCH_VERSATILE_AB
 	select CPU_32v4T
 	select CPU_ABRT_EV4T
 	select CPU_PABRT_LEGACY
@@ -167,7 +167,7 @@
 
 # ARM926T
 config CPU_ARM926T
-	bool "Support ARM926T processor" if ARCH_INTEGRATOR || MACH_REALVIEW_EB
+	bool "Support ARM926T processor" if ARCH_INTEGRATOR || MACH_REALVIEW_EB || ARCH_VERSATILE_PB || ARCH_VERSATILE_AB
 	select CPU_32v5
 	select CPU_ABRT_EV5TJ
 	select CPU_PABRT_LEGACY
@@ -175,6 +175,7 @@
 	select CPU_CP15_MMU
 	select CPU_COPY_V4WB if MMU
 	select CPU_TLB_V4WBI if MMU
+	depends on !CPU_V6 && !CPU_V7
 	help
 	  This is a variant of the ARM920.  It has slightly different
 	  instruction sequences for cache and TLB operations.  Curiously,
diff -ruN linux/arch/arm/mach-versatile/Kconfig linux.bak/arch/arm/mach-versatile/Kconfig
--- linux/arch/arm/mach-versatile/Kconfig	2011-02-07 18:03:55.000000000 -0600
+++ linux.bak/arch/arm/mach-versatile/Kconfig	2011-02-09 20:50:14.095413003 -0600
@@ -3,7 +3,6 @@
 
 config ARCH_VERSATILE_PB
 	bool "Support Versatile Platform Baseboard for ARM926EJ-S"
-	select CPU_ARM926T
 	select MIGHT_HAVE_PCI
 	default y
 	help
@@ -12,9 +11,16 @@
 
 config MACH_VERSATILE_AB
 	bool "Support Versatile Application Baseboard for ARM926EJ-S"
-	select CPU_ARM926T
 	help
 	  Include support for the ARM(R) Versatile Application Baseboard
 	  for the ARM926EJ-S.
 
+if ARCH_VERSATILE_PB || ARCH_VERSATILE_AB
+
+config ARCH_SUPPORTS_BIG_ENDIAN
+	bool
+	default y
+
+endif
+
 endmenu
